crates/core_simd/tests/pointers.rs

```diff
@@ -0,0 +1,59 @@
+#![feature(portable_simd, strict_provenance)]
+
+use core_simd::{Simd, SimdConstPtr, SimdMutPtr};
+
+macro_rules! common_tests {
+    { $constness:ident } => {
+        test_helpers::test_lanes! {
+            fn is_null<const LANES: usize>() {
+                test_helpers::test_unary_mask_elementwise(
+                    &Simd::<*$constness (), LANES>::is_null,
+                    &<*$constness ()>::is_null,
+                    &|_| true,
+                );
+            }
+
+            fn addr<const LANES: usize>() {
+                test_helpers::test_unary_elementwise(
+                    &Simd::<*$constness (), LANES>::addr,
+                    &<*$constness ()>::addr,
+                    &|_| true,
+                );
+            }
+
+            fn wrapping_offset<const LANES: usize>() {
+                test_helpers::test_binary_elementwise(
+                    &Simd::<*$constness (), LANES>::wrapping_offset,
```

Comment:
testing `wrapping_offset` on a `*mut ()` is kinda pointless since it always returns the input pointer, since `()` is a ZST. imho it would be better to test on `*mut u32` since it has both a nonzero pointee size (avoiding the output always being the same as the input), and a non-one pointee size (catching the potential bug where rustc forgot to multiply the offset by the pointee size)